cypress-io/cypress
# fix: mount component in [data-cy-root]

- Closes #24385 

### Additional details
Tweaks the Injectable `TestComponentRenderer` to inject the mounted Angular component into the standard `[data-cy-root]` element. Preserves the `rootId` that was being attached so as to avoid a breaking change if someone was relying on this selector (plus internal Angular Testbed still relies on this rootId existing).

### Steps to test
Throw a `.only` on https://github.com/cypress-io/cypress/blob/4bf516aef3cc12741921186071840e5e4fc1286a/system-tests/test/component_testing_spec.ts#L122

You can limit the version iteration so as to only test a single Angular version. Once it passes and you want to see the app itself, open up the scaffolded system test with `yarn cypress:open --project <path-tol-scaffolded-system-test>`

### How has the user experience changed?
<img width="801" alt="Screen Shot 2023-02-14 at 9 44 52 AM" src="https://user-images.githubusercontent.com/25158820/218788229-3d755908-bd41-42a3-ab8e-2a5bf5889340.png">


### PR Tasks

- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
